Generate.py

```diff
@@ -120,7 +120,7 @@ def main(args=None, callback=ERmain):
                 raise ValueError(f"File {fname} is invalid. Please fix your yaml.") from e
 
     # sort dict for consistent results across platforms:
-    weights_cache = {key: value for key, value in sorted(weights_cache.items())}
+    weights_cache = {key: value for key, value in sorted(weights_cache.items(), key=lambda k: k[0].lower())}
```

Comment:
Is the idea here that we're doing case-insensitive sorting only on the first letter, because it's unlikely to matter on the other letters?

If so, I'd rather do the whole string; it seems unlikely that sorting the filenames is a hot path (if it was, we probably couldn't afford to log all of them). [The Python docs suggest](https://docs.python.org/3/howto/sorting.html#key-functions) `key=str.casefold` as a way to do case-insensitive sorting, which seems more self-explanatory to me.

Reply:
The sort is operating on tuples, such as the below:
```
('user.yaml',
 ({'game': {'accessibility': 'items',
                               'progression_balancing': 0},
   'game': 'game',
   'name': 'user'},))
```

The filename (what we're trying to sort) is the zeroeth portion of that tuple. I wasn't able to use `str.lower()` because it's not a simple string, so had to use a lambda expression instead. I have made the change from lower() to casefold().

Reply:
Huh, I could've sworn I checked that this wasn't a tuple and that's why the [0] was confusing me. Oh well.